src/main/java/com/exasol/containers/status/ContainerStatusCache.java

```diff
@@ -0,0 +1,108 @@
+package com.exasol.containers.status;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Cache that remembers the container status across multiple test runs when the container is reused.
+ */
+public class ContainerStatusCache {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ContainerStatusCache.class);
+    static final String CACHE_SUB_DIRECTORY = "exasol_testcontainers";
+    static final String CACHE_FILE_EXTENSION = ".cache";
+    private final Path cacheDirectory;
+
+    /**
+     * Create a new instance of a {@link ContainerStatusCache}.
+     *
+     * @param tempDirectory temporary directory where cache files are stored
+     */
+    public ContainerStatusCache(final Path tempDirectory) {
+        this.cacheDirectory = tempDirectory.resolve(CACHE_SUB_DIRECTORY);
+    }
+
+    /**
+     * Get the directory in which the cache files are stored.
+     *
+     * @return cache directory
+     */
+    public Path getCacheDirectory() {
+        return this.cacheDirectory;
+    }
+
+    /**
+     * Check if a cache for a given container ID is available
+     *
+     * @param containerId docker container ID
+     * @return {@code true} if a cache is available for this container
+     */
+    public boolean isCacheAvailable(final String containerId) {
+        final Path cacheFile = getCacheFileForContainer(containerId);
+        return Files.exists(cacheFile);
+    }
+
+    private Path getCacheFileForContainer(final String containerId) {
+        return this.cacheDirectory.resolve(containerId + CACHE_FILE_EXTENSION);
+    }
+
+    /**
+     * Get the container state cache for the given container ID.
+     *
+     * @param containerId ID of the docker container for which the cache is requested.
+     * @return container state cache
+     */
+    public ContainerStatus read(final String containerId) {
+        return readFromCacheFile(getCacheFileForContainer(containerId));
+    }
+
+    private ContainerStatus readFromCacheFile(final Path cacheFile) {
+        LOGGER.debug("Reading container state from cache file \"{}\".", cacheFile);
+        try (final FileInputStream inputStream = new FileInputStream(cacheFile.toFile()); //
+                final ObjectInputStream objectInputStream = new ObjectInputStream(inputStream) //
+        ) {
+            return (ContainerStatus) objectInputStream.readObject();
+        } catch (final IOException | ClassNotFoundException exception) {
+            throw new ContainerStatusCacheException(
```

Comment:
How about using error-reporting-java for newly added exceptions?
I think it's not an issues if we have exceptions with and without error codes.

Reply:
Yes. I want to refactor that. But first I want to get this particularly long-running PR out of the way.